dll/win32/netcfgx/lang/en-US.rc

```diff
@@ -46,56 +46,56 @@ BEGIN
     CONTROL "", IDC_DNS2, "SysIPAddress32", WS_TABSTOP, 144, 165, 85, 12
 END
 
-IDD_TCPIP_ADVIP_DLG DIALOGEX 0, 0, 247, 247
+IDD_TCPIP_ADVIP_DLG DIALOGEX 0, 0, 247, 220
 STYLE DS_SHELLFONT | WS_CHILD | WS_CAPTION
 CAPTION "IP Settings"
 FONT 8, "MS Shell Dlg"
 BEGIN
-    GROUPBOX "IP addresses", -1, 5, 5, 240, 90
-    CONTROL "", IDC_IPLIST, "SysListView32", LVS_REPORT | LVS_SINGLESEL | LVS_SHOWSELALWAYS | LVS_NOSORTHEADER | WS_BORDER | WS_TABSTOP, 15, 15, 215, 55
-    PUSHBUTTON "Add...", IDC_IPADD, 60, 75, 50, 14, WS_TABSTOP
-    PUSHBUTTON "Edit...", IDC_IPMOD, 120, 75, 50, 14, WS_TABSTOP
-    PUSHBUTTON "Remove", IDC_IPDEL, 180, 75, 50, 14, WS_TABSTOP
-    GROUPBOX "Default gateways:", -1, 5, 100, 240, 90
-    CONTROL "", IDC_GWLIST, "SysListView32", LVS_REPORT | LVS_SINGLESEL | LVS_SHOWSELALWAYS | LVS_NOSORTHEADER | WS_BORDER | WS_TABSTOP, 15, 110, 215, 55
-    PUSHBUTTON "Add...", IDC_GWADD, 60, 170, 50, 14, WS_TABSTOP
-    PUSHBUTTON "Edit...", IDC_GWMOD, 120, 170, 50, 14, WS_TABSTOP
-    PUSHBUTTON "Remove", IDC_GWDEL, 180, 170, 50, 14, WS_TABSTOP
-    GROUPBOX "", -1, 5, 200, 240, 30
-    CHECKBOX "Automatic metric", IDC_AUTOMETRIC, 9, 200, 90, 12, BS_AUTOCHECKBOX | WS_TABSTOP
-    LTEXT "Interface metric:", -1, 15, 215, 90, 12
-    EDITTEXT IDC_METRIC, 110, 212, 50, 12, WS_TABSTOP | ES_NUMBER
+    GROUPBOX "IP addresses", -1, 5, 5, 240, 83
+    CONTROL "", IDC_IPLIST, "SysListView32", LVS_REPORT | LVS_SINGLESEL | LVS_SHOWSELALWAYS | LVS_NOSORTHEADER | WS_BORDER | WS_TABSTOP, 15, 15, 215, 50
+    PUSHBUTTON "Add...", IDC_IPADD, 60, 67, 50, 14, WS_TABSTOP
+    PUSHBUTTON "Edit...", IDC_IPMOD, 120, 67, 50, 14, WS_TABSTOP
+    PUSHBUTTON "Remove", IDC_IPDEL, 180, 67, 50, 14, WS_TABSTOP
+    GROUPBOX "Default gateways:", -1, 5, 93, 240, 83
+    CONTROL "", IDC_GWLIST, "SysListView32", LVS_REPORT | LVS_SINGLESEL | LVS_SHOWSELALWAYS | LVS_NOSORTHEADER | WS_BORDER | WS_TABSTOP, 15, 103, 215, 50
+    PUSHBUTTON "Add...", IDC_GWADD, 60, 155, 50, 14, WS_TABSTOP
+    PUSHBUTTON "Edit...", IDC_GWMOD, 120, 155, 50, 14, WS_TABSTOP
+    PUSHBUTTON "Remove", IDC_GWDEL, 180, 155, 50, 14, WS_TABSTOP
+    GROUPBOX "", -1, 5, 181, 240, 30
+    CHECKBOX "Automatic metric", IDC_AUTOMETRIC, 9, 179, 90, 12, BS_AUTOCHECKBOX | WS_TABSTOP
+    LTEXT "Interface metric:", -1, 15, 195, 90, 12
+    EDITTEXT IDC_METRIC, 110, 193, 50, 12, WS_TABSTOP | ES_NUMBER
 END
 
-IDD_TCPIP_ADVDNS_DLG DIALOGEX 0, 0, 247, 270
+IDD_TCPIP_ADVDNS_DLG DIALOGEX 0, 0, 247, 220
 STYLE DS_SHELLFONT | WS_CHILD | WS_CAPTION
 CAPTION "DNS"
 FONT 8, "MS Shell Dlg"
 BEGIN
-    LISTBOX IDC_DNSADDRLIST, 5, 17, 180, 50, LBS_NOTIFY
+    LISTBOX IDC_DNSADDRLIST, 5, 17, 180, 34, LBS_NOTIFY | LBS_NOINTEGRALHEIGHT
```

Comment:
Why this addition?

Reply:
3 reasons:
1.) it allows to define a height with exact pixel count, otherwise it increases in steps depending on multiples of the font height
2.) Windows uses LBS_NOINTEGRALHEIGHT as well in this control, I double-checked that
3.) It harmonizes to 100% the height displayed for this control in ResHacker (ros non-runtime) and ros-timetime

Reply:
ok, so this means that it should display a vertical scroll too or something as soon as an item gets partially shown at the bottom.

Reply:
Are there other flags to achieve that, or is that the default? Btw: I think this topic is academical. I have never seen more than one or 2 entries in both of those listboxes in my 22 years of Windows XP usage in private and at PCs at my workplace, or anywhere else in the world. If this would really need us to invest more love, that could also be done later imho.
But I will leave that conversation unresolved to keep its visibility.

Reply:
If you don't set the LBS_NOINTEGRALHEIGHT then windows will auto-resize the listbox so as to not show partial-items. This is at least [what the MSDN says](https://learn.microsoft.com/en-us/windows/win32/controls/list-box-styles)...
> Specifies that the size of the list box is exactly the size specified by the application when it created the list box. Normally, the system sizes a list box so that the list box does not display partial items.
For list boxes with the LBS_OWNERDRAWVARIABLE style, the LBS_NOINTEGRALHEIGHT style is always enforced.

(And here we don't use an owner-drawn listbox.)

Reply:
Ok, but that explains only what I wrote above already. It doesn't tell anything about any effect it would have on potential scrollbars. I have not tested it, but I would assume that we do get a scrollbar in both cases, whether we do set LBS_NOINTEGRALHEIGHT and also if we don't set it.
That moment just arrives sooner or later depending on the boxes height. And the flag does affect the boxes height.
iirc in comctl-v5-days all Listboxes did behave as if LBS_NOINTEGRALHEIGHT was set.
in v6 that can be controlled now. And by default v6 doesn't have it set, but tries to use multiples of the text height.
Obviously we do work in v6-context here.